Cargo.toml

```diff
@@ -13,11 +13,12 @@ categories = ["embedded", "hardware-support", "no-std"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-cortex-m = "0.7.5"
+cortex-m = { version = "0.7.5", features = ["critical-section-single-core"] }
```

Comment:
What is this feature needed for? I initially had a build error that I think is due to this feature being added in a later version of cortex-m.

Reply:
Before adding that I got errors that `_critical_section_1_0_release` and `_critical_section_1_0_acquire` were undefined.
```
wink@3900x 24-01-25T17:29:44.278Z:~/prgs/rust/forks/nucleo-f446re (Use-probe-rs)
$ cargo run --example blinky
   Compiling cortex-m v0.7.7
   Compiling critical-section v1.1.2
   Compiling rtt-target v0.4.0
   Compiling stm32f4 v0.14.0
   Compiling panic-probe v0.3.1
   Compiling nucleo-f446re v0.1.0 (/home/wink/prgs/rust/forks/nucleo-f446re)
   Compiling stm32f4xx-hal v0.13.2
error: linking with `rust-lld` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/wink/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:

...

-f446re/target/thumbv7em-none-eabi/debug/examples/blinky-246071663e38f089" "--gc-sections" "--nmagic" "-Tlink.x"
  = note: rust-lld: error: undefined symbol: _critical_section_1_0_acquire
          >>> referenced by lib.rs:183 (/home/wink/.cargo/registry/src/index.crates.io-6f17d22bba15001f/critical-section-1.1.2/src/lib.rs:183)
          >>>               rtt_target-50579e5237c22b68.rtt_target.d2ed4632eeeaf5a9-cgu.0.rcgu.o:(critical_section::with::h4960bdd999319092) in archive /home/wink/prgs/rust/forks/nucleo-f446re/target/thumbv7em-none-eabi/debug/deps/librtt_target-50579e5237c22b68.rlib
          
          rust-lld: error: undefined symbol: _critical_section_1_0_release
          >>> referenced by lib.rs:200 (/home/wink/.cargo/registry/src/index.crates.io-6f17d22bba15001f/critical-section-1.1.2/src/lib.rs:200)
          >>>               rtt_target-50579e5237c22b68.rtt_target.d2ed4632eeeaf5a9-cgu.0.rcgu.o:(core::ptr::drop_in_place$LT$critical_section..with..Guard$GT$::h5caa149425e17113) in archive /home/wink/prgs/rust/forks/nucleo-f446re/target/thumbv7em-none-eabi/debug/deps/librtt_target-50579e5237c22b68.rlib
          

error: could not compile `nucleo-f446re` (example "blinky") due to previous error
wink@3900x 24-01-25T17:30:37.535Z:~/prgs/rust/forks/nucleo-f446re (Use-probe-rs)
```

I did a search for [undefined _critical_section_1_0_release](https://www.google.com/search?q=undefined+_critical_section_1_0_release) and found [this](https://ferrous-systems.com/blog/defmt-rtt-linker-error/) so I added `critical-section-single-core`. Actually, I didn't find that ferrous post originally, but it pedantically explains why so used that one here :)

Reply:
Ah! I see what's going on now. I had an older lockfile initially, which only use 0.7.5, which does fail to build with the feature (because it was added in 0.7.6). But, newer versions of the dependencies use 0.7.7 which *does* have the feature, and requires it in this situation (as your link to the Ferrous Systems blog explains). At least it makes sense now.